src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -1985,14 +1985,13 @@ void GameObject::ModifyHealth(int32 change, Unit* attackerOrHealer /*= NULL*/, u
     // dealing damage, send packet
     if (player)
     {
-        WorldPacket data(SMSG_DESTRUCTIBLE_BUILDING_DAMAGE, 8 + 8 + 8 + 4 + 4);
-        data << GetPackGUID();
-        data << attackerOrHealer->GetPackGUID();
-        data << player->GetPackGUID();
-        data << uint32(-change);                    // change  < 0 triggers SPELL_BUILDING_HEAL combat log event
-                                                    // change >= 0 triggers SPELL_BUILDING_DAMAGE event
-        data << uint32(spellId);
-        player->SendDirectMessage(&data);
+        WorldPackets::GameObject::DestructibleBuildingDamage packet;
+        packet.Caster = player->GetGUID();
+        packet.Target = GetGUID();
+        packet.Damage = -change;
+        packet.Owner = attackerOrHealer->GetGUID(); // todo: this can be a GameObject
```

Comment:
Well, it is possible that fields are swapped in wpp, the order in structures from magic dump isnt absolute - it is only guaranteed to work for that build